docs/changelog.md

```diff
@@ -27,6 +27,54 @@ use the latest versions of the software.
 
 ## Release Notes
 
+### Version 0.6.0 (Release Date: X, 2020)
+
+```xml tab='Maven'
+<dependency>
+    <groupId>org.janusgraph</groupId>
+    <artifactId>janusgraph-core</artifactId>
+    <version>0.6.0</version>
+</dependency>
+```
+
+```groovy tab='Gradle'
+compile "org.janusgraph:janusgraph-core:0.6.0"
+```
+
+**Tested Compatibility:**
+
+* Apache Cassandra 2.2.10, 3.0.14, 3.11.0
+* Apache HBase 1.2.6, 1.3.1, 1.4.10, 2.1.5
+* Google Bigtable 1.3.0, 1.4.0, 1.5.0, 1.6.0, 1.7.0, 1.8.0, 1.9.0, 1.10.0, 1.11.0
+* Oracle BerkeleyJE 7.5.11
+* Elasticsearch 6.0.1, 6.6.0, 7.6.1
+* Apache Lucene 7.0.0
+* Apache Solr 7.0.0
+* Apache TinkerPop 3.4.6
+* Java 1.8
+
+For more information on features and bug fixes in 0.6.0, see the GitHub milestone:
+
+-   <https://github.com/JanusGraph/janusgraph/milestone/17?closed=1>
+
+#### Upgrade Instructions
+
+##### Serialization JanusGraphPredicates has changed
+
+GraphSON and Gryo Serialization has changed so both version need to be aligned.
```

Comment:
I commented on this already where you disabled a test that seems to be related to this, but this really needs some more explanation in my opinion to make it clear that serialization won't work between client and server if either is still on a version lower than 0.6.0 when the other is already updated to 0.6.0.

How about something like this:

>The serialization of JanusGraph predicates has changed in this version for both GraphSON and Gryo. It is therefore necessary to update both the client and the server to this version in parallel as the server will not be able to deserialize a JanusGraph predicate that was serialized by a client prior to version 0.6.0 once it was updated to version 0.6.0.